static/src/javascripts/projects/common/modules/article/spacefinder.js

```diff
@@ -187,70 +169,87 @@ define([
     // Rather than calling this directly, use spaceFiller to inject content into the page.
     // SpaceFiller will safely queue up all the various asynchronous DOM actions to avoid any race conditions.
     function findSpace(rules) {
-        var body;
+        var body, getDimensions;
 
-        rules = rules || defaultRules;
+        rules || (rules = defaultRules);
         body = rules.bodySelector ? document.querySelector(rules.bodySelector) : document;
+        getDimensions = rules.absoluteMinAbove ? _mapElementToComputedDimensions : _mapElementToDimensions;
 
         return getReady(body)
-        .then(getSlots)
+        .then(getCandidates)
         .then(getMeasurements)
         .then(enforceRules)
-        .then(filterSlots)
-        .then(returnSlots);
+        .then(returnCandidates);
 
-        function getSlots() {
-            var slots = qwery(rules.bodySelector + rules.slotSelector);
+        function getCandidates() {
+            var candidates = qwery(rules.bodySelector + rules.slotSelector);
             if (rules.fromBottom) {
-                slots.reverse();
+                candidates.reverse();
             }
             if (rules.startAt) {
                 var drop = true;
-                slots = filter(slots, function (slot) {
-                    if (slot === rules.startAt) {
+                candidates = candidates.filter(function (candidate) {
+                    if (candidate === rules.startAt) {
                         drop = false;
                     }
                     return !drop;
                 });
             }
             if (rules.stopAt) {
                 var keep = true;
-                slots = filter(slots, function (slot) {
-                    if (slot === rules.stopAt) {
+                candidates = candidates.filter(function (candidate) {
+                    if (candidate === rules.stopAt) {
                         keep = false;
                     }
                     return keep;
                 });
             }
-            return slots;
+            return candidates;
         }
 
-        function getMeasurements(slots) {
+        function getMeasurements(candidates) {
+            var contentMeta = rules.clearContentMeta ?
+                document.querySelector('.js-content-meta') :
+                null;
+            var opponents = rules.selectors ?
```

Comment:
It's a quibble, but perhaps this array should be flattened right from the off? It's a triviality, though.